File: jambeth/jAmbeth.DataChange/src/de/osthus/ambeth/datachange/model/IDataChangeEntry.java
package de.osthus.ambeth.datachange.model;

import de.osthus.ambeth.annotation.XmlType;

@XmlType
public interface IDataChangeEntry
{
	Class<?> getEntityType();

	Object getId();

	byte getIdNameIndex();

	Object getVersion();

	String[] getTopics();

	void setTopics(String[] topics);
}

File: ambeth/Ambeth.DataChange/ambeth/datachange/model/IDataChangeEntry.cs
using System;
using De.Osthus.Ambeth.Annotation;

namespace De.Osthus.Ambeth.Datachange.Model
{
    [XmlType(Name = "IDataChangeEntry", Namespace = "http://schemas.osthus.de/Ambeth")]
    public interface IDataChangeEntry
    {
        Type EntityType { get; }

        sbyte IdNameIndex { get; }

        Object Id { get; }

        Object Version { get; }

        String[] Topics { get; set; }
    }
}
